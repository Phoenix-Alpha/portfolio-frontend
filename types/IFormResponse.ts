export interface IFormResponse {
  isSuccess: boolean
  message?: string
}
