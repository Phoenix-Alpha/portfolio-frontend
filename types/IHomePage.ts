export interface IHomePage {
  heroContent?: string
}
