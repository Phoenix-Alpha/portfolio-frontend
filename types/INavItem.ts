export interface INavItem {
  key: string
  title: string
  url?: string
  isGroup?: boolean
  divider?: boolean
  subItems?: INavItem[]
  external?: string
  target?: string
}
