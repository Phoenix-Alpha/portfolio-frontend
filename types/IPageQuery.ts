import { DocumentNode } from 'graphql'
import { AnyAction } from 'redux'

export interface IPageQuery {
  query: DocumentNode
  variables?: Record<string, unknown>
  context?: Record<string, unknown>
  path?: string
  normalizer?: (pathEntity?: unknown) => unknown
  action?: (pathEntity?: unknown) => AnyAction
  actions?: Array<(pathEntity?: unknown) => AnyAction>
}
