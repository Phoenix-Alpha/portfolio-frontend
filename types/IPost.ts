export interface IPost {
  id: string
  title: string
  description: string
  slug: string
  date?: string
  cover?: string
  tags?: string[]
  readTimeMins?: number
  content?: string
}
