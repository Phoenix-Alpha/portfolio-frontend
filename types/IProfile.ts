export interface IProfile {
  firstName: string
  lastName: string
  email: string
  headline?: string
  bio?: string
  avatar?: string
  country?: string
  state?: string
  city?: string
  address?: string
}
