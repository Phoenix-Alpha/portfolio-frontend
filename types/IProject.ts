export interface IProject {
  id: string
  title: string
  description: string
  slug: string
  link?: string
  cover?: string
  tags?: string[]
  content?: string
}
