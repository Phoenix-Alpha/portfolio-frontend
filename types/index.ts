export * from './pages'
export * from './IRawUploadFile'
export * from './IPageQuery'
export * from './IProfile'
export * from './IStoreState'
export * from './TTheme'
export * from './INavItem'
export * from './IFormResponse'
