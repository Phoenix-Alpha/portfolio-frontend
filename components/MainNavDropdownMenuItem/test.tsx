import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { INavItem } from '@/types'

import { MainNavDropdownMenuItem } from '.'

describe('<MainNavDropdownMenuItem />', () => {
  it('should renders correct <MainNavDropdownMenuItem />', () => {
    const item: INavItem = {
      key: 'test-key',
      title: 'test-tilte'
    }

    const wrapper = shallow(<MainNavDropdownMenuItem item={item} />)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
