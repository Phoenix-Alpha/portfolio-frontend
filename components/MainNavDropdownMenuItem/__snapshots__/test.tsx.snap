// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainNavDropdownMenuItem /> should renders correct <MainNavDropdownMenuItem /> 1`] = `
<li
  className="dropdown-menu-item hover:bg-[#EFF1FC]"
  key="test-tilte"
  onClick={[Function]}
>
  <Link>
    <a
      className="block py-3 px-6 xl:px-4 w-full"
    >
      test-tilte
    </a>
  </Link>
</li>
`;
